nmt.go

```diff
@@ -128,9 +129,18 @@ func New(h hash.Hash, setters ...Option) *NamespacedMerkleTree {
 	for _, setter := range setters {
 		setter(opts)
 	}
-	treeHasher := NewNmtHasher(h, opts.NamespaceIDSize, opts.IgnoreMaxNamespace)
+
+	// first create the default hasher using the updated options
+	hasher := NewNmtHasher(h, opts.NamespaceIDSize, opts.IgnoreMaxNamespace)
+	opts.Hasher = hasher
+
+	// set the options a second time to replace the hasher if needed
+	for _, setter := range setters {
+		setter(opts)
+	}
```

Comment:
I hate doing this a second time, but I'm not sure what else to do without sacrificing expected behavior from the user. 

we can't set hasher using the default namespace id size, and then the user also be able to change the default namespace ID size. So we have to set the options first, then create the default hasher, then optionally replace the default hasher with a user specified one